src/common/components/CheckboxAggregation.jsx

```diff
@@ -23,37 +27,66 @@ class CheckboxAggregation extends Component {
 
   constructor(props) {
     super(props);
-    this.state = { selectionMap: Immutable.Map() };
+    this.state = {
+      selectionMap: Immutable.Map(),
+      maxBucketCountToDisplay: BUCKET_CHUNK_SIZE,
+    };
+
+    this.onShowMoreClick = this.onShowMoreClick.bind(this);
   }
 
   onSelectionChange(key, selected) {
     let { selectionMap } = this.state;
     selectionMap = selectionMap.set(key, selected);
     this.setState({ selectionMap });
-    const selections = selectionMap.keySeq()
+    const selections = selectionMap
+      .keySeq()
       .filter(bucketKey => selectionMap.get(bucketKey))
       .toArray();
     this.props.onChange(selections);
   }
 
+  onShowMoreClick() {
+    const { maxBucketCountToDisplay } = this.state;
+    this.setState({
+      maxBucketCountToDisplay: maxBucketCountToDisplay + BUCKET_CHUNK_SIZE,
+    });
+  }
+
+  renderShowMore() {
+    const { buckets } = this.props;
+    const { maxBucketCountToDisplay } = this.state;
+
+    if (maxBucketCountToDisplay >= buckets.size) {
+      return null;
+    }
+
+    return <Button onClick={this.onShowMoreClick}>Show more</Button>;
+  }
+
   render() {
+    const { maxBucketCountToDisplay, selectionMap } = this.state;
     return (
       <div style={{ width: '100%' }}>
-        <List header={<strong>{this.props.name}</strong>}>
-          {this.props.buckets.map(bucket => (
+        <List
+          header={<strong>{this.props.name}</strong>}
+          footer={this.renderShowMore()}
+        >
+          {this.props.buckets.take(maxBucketCountToDisplay).map(bucket => (
             <List.Item key={bucket.get('key')}>
-              <List.Item.Meta title={
-                <CheckboxItem
-                  checked={this.state.selectionMap.get(bucket.get('key'))}
-                  onChange={(checked) => { this.onSelectionChange(bucket.get('key'), checked); }}
-                >
-                  {bucket.get('key')}
-                </CheckboxItem>
-              }
+              <List.Item.Meta
+                title={
+                  <CheckboxItem
+                    checked={selectionMap.get(bucket.get('key'))}
+                    onChange={checked => {
```

Comment:
I just realized that we have patterns likes this, it's better to avoid it 

https://reactjs.org/docs/faq-functions.html#arrow-function-in-render

Reply:
yep unfortunately we have few, this change came from prettier but I'll still fix that.

Reply:
ah but in that case, there is no easy to around it because we need to pass what `onChange` emits and bucket key which is defined only inside the `map` loop.